index.bs

```diff
@@ -1370,77 +880,818 @@ What configuration is the device in after it resets?
+slots</a> described in the following table:
+
+<table class="data" dfn-for="USBDevice" dfn-type="attribute">
+  <tr>
+    <th><a>Internal Slot</a></th>
+    <th>Initial Value</th>
+    <th>Description (non-normative)</th>
+  </tr>
+  <tr>
+    <td><dfn>\[[configurations]]</dfn></td>
+    <td>An empty <a>sequence</a> of {{USBConfiguration}}</code></td>
+    <td>
+      All configurations supported by this device.
+    </td>
+  </tr>
+  <tr>
+    <td><dfn>\[[configurationValue]]</dfn></td>
+    <td>&lt;always set in prose></td>
+    <td>
+      The current configuration value of the device.
+    </td>
+  </tr>
+  <tr>
+    <td><dfn>\[[selectedAlternateSetting]]</dfn></td>
+    <td>An <a>empty</a> list of <a>integer</a></td>
+    <td>
+      The current alternate setting for each interface on the current configuration.
+    </td>
+  </tr>
+  <tr>
+    <td><dfn>\[[claimedInterface]]</dfn></td>
+    <td>An <a>empty</a> list of <a>boolean</a></td>
+    <td>
+      The claimed status for each interface on the current configuration.
+    </td>
+  </tr>
+</table>
+
+<div algorithm="To find the device descriptor for the connected usb device">
+    <dfn>To find the device descriptor for the connected usb device</dfn>,
+    performing the following steps:
+    1. Let |deviceDescriptor| be the <a>device descriptor</a> of the connected
+        device by performing <a>Get Descriptor</a> with
+        <code>DescriptorType</code> set to <code>DEVICE</code>.
+    1. Return |deviceDescriptor|.
+</div>
+
+<div algorithm="To find a list of configuration descriptors for the connected usb device">
+    <dfn>To find a list of configuration descriptors for the connected usb device</dfn>,
+    performing the following steps:
+    1. Let |deviceDescriptor| be the result of
+        <a>to find the device descriptor for the connected usb device</a>.
+    1. Let |numConfigurations| be <code>bNumConfigurations</code> of |deviceDescriptor|.
+    1. Let |configurationIndex| be 0.
+    1. Let |configurationDescriptors| be an <a>empty</a> <a>list</a>.
+    1. While |configurationIndex| is smaller than |numConfigurations|:
+        1. Let |descriptors| be the a list of <a>descriptors</a>
+            by performing <a>Get Descriptor</a> with
+            <code>DescriptorType</code> set to <code>CONFIGURATION</code> and
+            <code>DescriptorIndex</code> set to |configurationIndex|.
+        1. If the <code>bDescriptorType</code> of the |descriptors|[0] is equal to <code>CONFIGURATION</code>,
+            <a>append</a> |descriptors|[0] to |configurationDescriptors|.
+        1. Advance |configurationIndex| by 1.
+    1. Return |configurationDescriptors|.
+</div>
+
+<div algorithm="To abort transfers currently scheduled on an interface">
+    <dfn>To abort transfers currently scheduled on an interface</dfn> with the given |interface| object,
+   performing the following steps:
+    1. Let |device| be the target {{USBDevice}} object.
```

Comment:
There is no target `USBDevice` object for this algorithm. Add *device* as a second parameter to these steps.

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/1a96f493fc285da99d2b812d809c241388197beb.